code/modules/mob/living/carbon/human/species/xenomorphs/alien_embryo.dm

```diff
@@ -0,0 +1,154 @@
+		RemoveInfectionImages(affected_mob)
+	..()
+
+/obj/item/alien_embryo/process()
+	if(!affected_mob)	return
+	if(loc != affected_mob)
+		affected_mob.status_flags &= ~(XENO_HOST)
+		STOP_PROCESSING(SSprocessing, src)
+		RemoveInfectionImages(affected_mob)
+		affected_mob = null
+		return
+
+	if(stage < 5 && prob(3))
+		stage++
+		spawn(0)
+			RefreshInfectionImage(affected_mob)
+
+	switch(stage)
+		if(2, 3)
+			if(prob(1))
+				affected_mob.emote("sneeze")
+			if(prob(1))
+				affected_mob.emote("cough")
+			if(prob(1))
+				affected_mob  << "<span class='danger'>Your throat feels sore.</span>"
+			if(prob(1))
+				affected_mob  << "<span class='danger'>Mucous runs down the back of your throat.</span>"
+		if(4)
+			if(prob(1))
+				affected_mob.emote("sneeze")
+			if(prob(1))
+				affected_mob.emote("cough")
+			if(prob(2))
+				affected_mob  << "<span class='danger'>Your muscles ache.</span>"
+				if(prob(20))
+					affected_mob.take_organ_damage(1)
+			if(prob(2))
+				affected_mob  << "<span class='danger'>Your stomach hurts.</span>"
+				if(prob(20))
+					affected_mob.adjustToxLoss(1)
+					affected_mob.updatehealth()
+		if(5)
+			affected_mob  << "<span class='danger'>You feel something tearing its way out of your stomach!</span>"
+			affected_mob.adjustToxLoss(10)
+			affected_mob.updatehealth()
+			if(prob(50))
+				AttemptGrow()
+
+/obj/item/alien_embryo/proc/AttemptGrow(var/gib_on_success = 1)
+	var/list/candidates = get_alien_candidates()
+	var/picked = null
+
+	// To stop clientless larva, we will check that our host has a client
+	// if we find no ghosts to become the alien. If the host has a client
+	// he will become the alien but if he doesn't then we will set the stage
+	// to 2, so we don't do a process heavy check everytime.
+
+	if(candidates.len)
+		picked = pick(candidates)
+	else if(affected_mob.client)
+		picked = affected_mob.key
+	else
+		stage = 4 // Let's try again later.
+		return
+
+	if(affected_mob.lying)
+		affected_mob.overlays += image('icons/mob/alien.dmi', loc = affected_mob, icon_state = "burst_lie")
+	else
+		affected_mob.overlays += image('icons/mob/alien.dmi', loc = affected_mob, icon_state = "burst_stand")
+	spawn(6)
+		var/mob/living/carbon/alien/larva/new_xeno = new(affected_mob.loc)
+		new_xeno.key = picked
+		new_xeno << sound('sound/voice/hiss5.ogg',0,0,0,100)	//To get the player's attention
+		if(gib_on_success)
+			affected_mob.gib()
+		qdel(src)
+
+/*----------------------------------------
+Proc: RefreshInfectionImage()
+Des: Removes all infection images from aliens and places an infection image on all infected mobs for aliens.
+----------------------------------------*/
+/obj/item/alien_embryo/proc/RefreshInfectionImage()
+
+	for(var/mob/living/carbon/alien in player_list)
+
+		if(!locate(/obj/item/organ/xenos/hivenode) in alien.internal_organs)
+			continue
+
+		if(alien.client)
+			for(var/image/I in alien.client.images)
+				if(dd_hasprefix_case(I.icon_state, "infected"))
```

Comment:
Yeah, no. Find a better way than qdeling images, it's laggy and doesn't work.

Reply:
Not really sure how to handle the removal of those images right now.

Reply:
Should be fixed now.